src/vaadin-form-layout.html

```diff
@@ -470,29 +463,34 @@
           }
 
           let col = 0;
-          Array.from(this.children).forEach((child) => {
-            if (getComputedStyle(child).display !== 'none') {
-              let colspan;
+          Array.from(this.children).forEach((child, index) => {
+            if (child.localName === 'br') {
+              // Reset column count on line break
+              col = 0;
+            } else if (getComputedStyle(child).display !== 'none') {
+              const followingVisibleItems = Array.from(this.children)
+                .slice(index + 1)
+                .filter(item => item.localName === 'br' || getComputedStyle(item).display !== 'none');
+
+              const nextLineBreak = followingVisibleItems.length > 0 &&
```

Comment:
Seems that `followingVisibleItems` is only used to determine if the next item is a line break. This introduces a loop iterating all items inside the loop that already iterates all items, which is bad performance-wise.

How about filtering the array only once instead?

```javascript
let col = 0;
Array.from(this.children)
  .filter(child => child.localName === 'br' || getComputedStyle(child).display !== 'none')
  .forEach((child, index, children) => {
    if (child.localName === 'br') {
      // Reset column count on line break
      col = 0;
      return;
    }

    const nextIndex = index + 1;
    const nextLineBreak = nextIndex < children.length && children[nextIndex].localName === 'br';
    // ...
  });
```

Reply:
Done. Update the check for `nextLineBreak`.